Twilio channel to server side

fredrifo opened the pull request.

FredrikForsell committed:
Update twilioService.js

+ Added loop that checks for matching channels.
  - If it finds a match, open channel with channel.getchannel.
  - If not, send api call to create channel
+ subscribed to channelInvited event.
   - If user clicks ok -> joins channel

FredrikForsell committed:
Minor change.

Log requestUrl to console.

FredrikForsell committed:
Added a default message

FredrikForsell committed:
Moved Twilio logic to API side.

FredrikForsell committed:
Removed Console.log

FredrikForsell committed:
Create channel and add users

FredrikForsell committed:
Removed the default chat message.

FredrikForsell committed:
Removed code from testing

bassamseif commented on screens/HomeAuth.js:
```diff
@@ -41,6 +41,7 @@ class HomeAuth extends Component {
       `${API_URL}/register/${authType}?access_token=${token}`,
       { method: 'post' }
     );
+    // console.log(`${API_URL}/register/${authType}?access_token=${token}`);
```
probably forgotten console log?

bassamseif commented on services/twilioService.js:
```diff
@@ -59,8 +63,9 @@ class TwilioService {
-    } catch {
+    } catch (er) {
       // TODO: Add sentry logging
+      console.log(`Failed to initiate channel ${er}`);
```
maybe we can use `SET_ERROR` https://github.com/Hey-Mentor/mentor-mobile/blob/master/reducers/GeneralReducer.js#L7-L11

FredrikForsell committed:
Add Shared media link on profile

FredrikForsell committed:
Create LinkListItem.js

Custom list item that displays link, link title & FavIcon.

Before the link title has been fetched it will be shown as the domain name.

FredrikForsell committed:
Create LinkList.js

Custom Link list using sectionList.
Sections includes a divider with group name.

FredrikForsell committed:
Update yarn.lock

FredrikForsell committed:
Add shared media screen

FredrikForsell committed:
Add html scraper library

FredrikForsell committed:
Passing navigation prop

FredrikForsell committed:
Create SharedMediaScreen.js

fetch data and display in custom linkList
